docs/guides/ReadingMemoryOptimization.md

```diff
@@ -0,0 +1,321 @@
+
+## Overview
+
+APIs for reading Parquet files:
+1. **LogicalColumnReader API** - Column-oriented reading with type-safe access
+2. **Arrow API (FileReader)** - Row-oriented reading using Apache Arrow's in-memory format
+
+Each API offers different memory management options that impact memory usage.
+
+## Memory Configuration Parameters
+
+### 1. Buffer Size
+Controls the size of I/O buffers used when reading from disk or streams.
+
+**Default**: 8 MB (8,388,608 bytes) when using default file reading
+
+**Impact**: Larger buffers reduce I/O operations but increase memory usage. Smaller buffers are more memory-efficient but may decrease throughput.
+
+### 2. Chunked Reading
+Instead of loading entire columns into memory, read data in smaller chunks.
+
+**Impact**: Reduces peak memory usage by processing data incrementally.
+
+### 3. Pre-buffering (Arrow API Only)
+Controls whether the Arrow FileReader pre-fetches data from multiple row groups ahead of time.
+
+**Default**: `true`
+
+**Impact**: Pre-buffering can significantly increase memory usage as it loads data from future row groups before they're needed. This is the primary cause of memory usage scaling with file size reported in Apache Arrow [issue #46935](https://github.com/apache/arrow/issues/46935).
+
+### 4. Cache (Arrow API Only)
+The Arrow API uses an internal `ReadRangeCache` that stores buffers for column chunks.
+
+**Behavior**: Once loaded, buffers remain in cache and are not evicted, causing memory to accumulate while iterating through large files.
+
+**Impact**: Can cause memory usage to reach the total size of data being read.
+
+## 1. LogicalColumnReader API
+
+The @ParquetSharp.LogicalColumnReader`1 API provides direct, column-oriented access to Parquet data with efficient memory usage characteristics.
+
+### Basic Usage 
+
+```csharp
+public static void ReadWithLogicalReader(string filePath)
+{
+    // Open file with default settings
+    using var fileReader = new ParquetFileReader(filePath);
+    var fileMetadata = fileReader.FileMetaData;
+
+    for (int rg = 0; rg < metadata.NumRowGroups; rg++)
+    {
+        using var rowGroupReader = fileReader.RowGroup(rg);
+        long numRows = rowGroupReader.MetaData.NumRows;
+
+        for (int col = 0; col < metadata.NumColumns; col++)
+        {
+            using var columnReader = rowGroupReader.Column(col);
+            using var logicalReader = columnReader.LogicalReader<float>();
+
+            var values = new float[numRows];
+            logicalReader.ReadBatch(values);
+        }
+    }
+}
+```
+
+### Buffered Stream
+
+```csharp
+public static void ReadWithCustomBuffer(string filePath, int bufferSize = 1024 * 1024)
+{
+    // Use a buffered stream with custom buffer size (1 MB in this example)
+    using var fileStream = File.OpenRead(filePath);
+    using var bufferedStream = new BufferedStream(fileStream, bufferSize);
```

Comment:
By using a buffered stream, I actually meant enabling it in the ReaderProperties with `ReaderProperties.EnableBufferedStream`.

In previous investigations I've found that this can significantly reduce memory usage.

I don't think using a .NET System.IO.BufferedStream will change memory usage characteristics much.